src/graph/SequentialExecutor.cpp

```diff
@@ -19,10 +19,21 @@ SequentialExecutor::SequentialExecutor(SequentialSentences *sentences,
 }
 
 
+
 Status SequentialExecutor::prepare() {
     for (auto i = 0U; i < sentences_->sentences_.size(); i++) {
         auto *sentence = sentences_->sentences_[i].get();
         auto executor = makeExecutor(sentence);
+        if (FLAGS_enable_authorize) {
+            auto *session = executor->ectx()->rctx()->session();
+            /**
+             * Skip special operations check at here. they are :
+             * kUse, kDescribeSpace, kRevoke and kGrant.
+             */
+            if (!PermissionCheck::permissionCheck(session, sentence)) {
```

Comment:
What if the query is "use space test; sth else", can we work? But this is a quite usual.

Reply:
> What if the query is "use space test; sth else", can we work? But this is a quite usual.

Good question. you are right . it's fail when "use space ; sth else" . 
because our permission check is in the preparation phase,  have not set the space id in the session yet in the preparation phase. so the second statement should be fail.
I originally intended to check permissions in each executor, but this would result in too much code clutter.

Reply:
ok.